BJS2-25779

Caveri95 opened the pull request:
- Добавлен эндпоинт на получение AnalyticsEvent по receiverId и временному диапазону

Caveri95 committed:
BJS2-25779

- Добавлен эндпоинт на получение AnalyticsEvent по receiverId и временному диапазону

Caveri95 committed:
BJS2-25779

- Добавлен интеграционный тест на контроллер получения аналитики

Caveri95 committed:
BJS2-25779

- Добавлен тест валидатора

Caveri95 committed:
BJS2-25779

- Правки в тестах

pablo023 commented on src/main/java/faang/school/analytics/controller/AnalyticsEventController.java:
```diff
@@ -0,0 +1,37 @@
+
+    @GetMapping
+    public List<AnalyticEventDto> getAnalytics(@RequestParam long receiverId,
+                                               @RequestParam String eventType,
+                                               @RequestParam(required = false) String interval,
+                                               @RequestParam(required = false) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime from,
+                                               @RequestParam(required = false) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime to) {
```
Почему бы не сделать это дто? Как-то много параметров как будто, сложно выглядит, читается не очень

Caveri95 replied:
Да, хорошо

pablo023 commented on src/test/resources/db/data.sql:
Это видимо тестовые данные, которые случайно сюда попали

Caveri95 replied:
Это данные для заполнения БД для тестов. Использовал тестовую БД для тестов контроллера

pablo023 commented on src/test/resources/application.yaml:
```diff
@@ -0,0 +1,18 @@
+    password: ${DB_PASSWORD}
+  data:
+    redis:
+      host: ${spring.data.redis.host}
+      port: ${spring.data.redis.port}
+      channel:
+        profileView: profile_view_channel
+        followerView: follower_view_channel
```
Похоже на часть функционала из другой задачи, здесь это должно быть?

Caveri95 replied:
У меня PR получается в другую мою задачу, а там нужно было конфиг Redis делать, поэтому это здесь

pablo023 commented on src/test/resources/db/schema.sql:
А зачем нам эти эвенты в базе сторить? Объясни, пожалуйста.

Caveri95 replied:
Туда же к тестовым данным для тестов контроллеров

Caveri95 committed:
BJS2-25779

- Исправлен метод контроллера на Post, чтобы можно было передать тело запроса для получения аналитики
- Исправлены тесты
- Создан объект DTO для запроса аналитики

pablo023 commented:
Пофикси конфликты, и можешь мержить, и кидать в done

Caveri95 committed:
Merge branch 'titan-master-bc5-BJS2-25764' into titan-master-bc5-BJS2-25779

# Conflicts:
#	src/main/java/faang/school/analytics/dto/AnalyticInfoDto.java
#	src/main/java/faang/school/analytics/model/Interval.java
#	src/main/java/faang/school/analytics/service/AnalyticsEventService.java
#	src/test/java/faang/school/analytics/service/AnalyticsEventServiceTest.java

Caveri95 committed:
BJS2-25779

- conflict dix
